### PR TITLE
feat: plugin define accept `undefined`

### PR DESCRIPTION
**Pull request by Asuka109**
## Description

Refer to https://github.com/modern-js-dev/modern.js/issues/2692

## Types of changes

- [ ] Docs change / Dependency upgrade
- [ ] Bug fix
- [ ] New feature / Improvement
- [ ] Refactoring
- [ ] Breaking change

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/builder/builder-webpack-provider/src/plugins/define.ts [hunk 1]
```diff
@@ -9,13 +9,14 @@ export const builderPluginDefine = (): BuilderPlugin => ({
       const config = api.getNormalizedConfig();
```
Comment (chenjiahan) [inline packages/builder/builder-webpack-provider/src/plugins/define.ts:9]: 加个 changeset 哈

File: packages/builder/builder-webpack-provider/src/plugins/define.ts [hunk 2]
```diff
@@ -9,13 +9,14 @@ export const builderPluginDefine = (): BuilderPlugin => ({
...
+        'process.env.NODE_ENV': process.env.NODE_ENV,
       };
 
       // Serialize global vars. User can customize value of `builtinVars`.
       const globalVars = { ...builtinVars, ...config.source.globalVars };
-      const serializedVars = _.mapValues(globalVars, value =>
-        JSON.stringify(value),
+      const serializedVars = _.mapValues(
+        globalVars,
+        value => JSON.stringify(value) ?? 'undefined',
```
Comment (chenjiahan) [inline packages/builder/builder-webpack-provider/src/plugins/define.ts:19]: null 也需要转成 undefined 吗

Comment (Asuka109) [reply packages/builder/builder-webpack-provider/src/plugins/define.ts:19]: null 可以正常被 json.stringify handle 的，这里只要处理 undefined 就好

Comment (chenjiahan) [reply packages/builder/builder-webpack-provider/src/plugins/define.ts:19]: ?? 会处理 null

Comment (Asuka109) [reply packages/builder/builder-webpack-provider/src/plugins/define.ts:19]: stringify 接收 `null` 的时候会输出字符串 `"null"`，所以这里应该可以的

Comment (chenjiahan) [reply packages/builder/builder-webpack-provider/src/plugins/define.ts:19]: 噢噢，也对

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**